src/ddsrt/src/string.c

```diff
@@ -219,3 +219,41 @@ ddsrt_strndup(const char *str, size_t len)
 
   return s;
 }
+
+void
+ddsrt_str_trim(char *str)
+{
+  int index, i;
+
+  if (str == NULL) {
+    return;
+  }
+  
+  /* trim leading white spaces */
+  index = 0;
+  while(str[index] == ' ' || str[index] == '\t' || str[index] == '\n') {
```

Comment:
I would expect a generic "trim whitespace" to rely on `isspace`, so also handle `'\r'`, `\v` and `\f`. Or (see comment below) strictly limit it to the ordinary space character (`' '`), but then I think the name/description should be updated to reflect that.